virttest/qemu_vm.py

```diff
@@ -4300,11 +4312,15 @@ def migrate(self, timeout=virt_vm.BaseVM.MIGRATE_TIMEOUT, protocol="tcp",
             # If we're doing remote migration and it's completed successfully,
             # self points to a dead VM object
             if not not_wait_for_migration:
-                if self.is_alive() and self.is_paused():
+                if self.is_alive():
                     # For short period of time the status can be "inmigrate"
                     # for example when using external program
                     # (qemu commit fe823b6f87b2ebedd692ca480ceb9693439d816e)
-                    self.resume(60)
+                    # resume vm during "inmigrate" status can't work
+                    if utils_misc.wait_for(lambda: not self.is_inmigrate(),
+                                           timeout=60, step=1):
+                        if self.is_paused():
+                            self.resume()
```

Comment:
in case it's still in `inmigrate` status you want to ignore it and silently proceed? I think an exception would be more appropriate so we can identify why it stayed in `inmigrate` status. In case it's common we should at least log a message...

Reply:
it's reasonable to raise exception if VM status is not "inmigrate" after timeout.